### ISSUE TITLE
EncapsulatePublicFieldInspection needs to ignore controls on UserForms

### ISSUE THREAD

**Issue by comintern** [rubberduck-vba/Rubberduck#2599]
Create a `UserForm` with a `CommandButton` name `CommandButton1` and run inspections (no code behind the form).  This is the result:

![untitled](https://cloud.githubusercontent.com/assets/8944005/22469664/6b41bd92-e792-11e6-8beb-e6c6c9bdc826.png)

I'm not *entirely* sure how this is handled internally, but my guess is that controls on `UserForm`s are accessible because `Controls` is the default member of the `_UserForm` interface:

```
dispinterface _UserForm {
    properties:
    methods:
        //...
        [id(00000000), propget, helpcontext(0x001e884a)]
        Controls* Controls();
```

**Comment by retailcoder** [comment]
This sounds very similar to #2446 - when we iterate the `Controls` collection, we create a `Declaration` of type `DeclarationType.Control`, with `Accessibility.Public`  visibility, under the form's scope.. so we essentially create a public field for every control; the inspection simply needs to ignore those.

https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Parsing/Symbols/DeclarationSymbolsListener.cs#L160-L191

The `<remarks>` comment is misleading: these declarations allow the context-selection to pick up a reference to a form control in the code-behind, not just identify event procedures.

````csharp
        /// <summary>
        /// Scans form designer to create a public, self-assigned field for each control on a form.
        /// </summary>
        /// <remarks>
        /// These declarations are meant to be used to identify control event procedures.
        /// </remarks>
        private void DeclareControlsAsMembers(IVBComponent form)
        {
            if (form.Controls == null) { return; }

            foreach (var control in form.Controls)
            {
                var typeName = control.TypeName();
                // The as type declaration should be TextBox, CheckBox, etc. depending on the type.
                var declaration = new Declaration(
                    _qualifiedName.QualifyMemberName(control.Name),
                    _parentDeclaration,
                    _currentScopeDeclaration,
                    string.IsNullOrEmpty(typeName) ? "Control" : typeName,
                    null,
                    true,
                    true,
                    Accessibility.Public,
                    DeclarationType.Control,
                    null,
                    Selection.Home,
                    false,
                    null,
                    false);
                AddDeclaration(declaration);
            }
        }
````

So the inspection simply needs to exclude `DeclarationType.Control` declarations.

**Comment by retailcoder** [comment]
fixed with d2ba220c6e5b62e1c07a9098aa347507acf5aed1
